packages/flutter/test/material/app_test.dart

```diff
@@ -447,7 +447,99 @@ void main() {
+    // Mock the Window to explicitly report a light platformBrightness.
+    tester.binding.window.platformBrightnessTestValue = Brightness.light;
+
+    ThemeData appliedTheme;
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+            brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        themeMode: ThemeMode.light,
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+    expect(appliedTheme.brightness, Brightness.light);
+
+    // Mock the Window to explicitly report a dark platformBrightness.
+    tester.binding.window.platformBrightnessTestValue = Brightness.dark;
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+            brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        themeMode: ThemeMode.light,
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+    expect(appliedTheme.brightness, Brightness.light);
+  });
+
+  testWidgets('MaterialApp uses darkTheme when themeMode is dark', (WidgetTester tester) async {
+    // Mock the Window to explicitly report a light platformBrightness.
+    tester.binding.window.platformBrightnessTestValue = Brightness.light;
+
+    ThemeData appliedTheme;
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+            brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        themeMode: ThemeMode.dark,
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+    expect(appliedTheme.brightness, Brightness.dark);
+
+    // Mock the Window to explicitly report a dark platformBrightness.
+    tester.binding.window.platformBrightnessTestValue = Brightness.dark;
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+            brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        themeMode: ThemeMode.dark,
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+    expect(appliedTheme.brightness, Brightness.dark);
+  });
+
+  testWidgets('MaterialApp uses regular theme when themeMode is system & platformBrightness is light', (WidgetTester tester) async {
```

Comment:
Is it possible that use of symbols like "&" in test names could cause complications for devs that try to execute specific tests via command line? Seems like there could be cases where this might require some kind of encoding, e.g., URL encoding, which would be an unnecessary burden...

Reply:
Good point, I will fix this.